include/particle.h

```diff
@@ -310,45 +262,47 @@ class Particle : public ParticleBase<Tdim> {
   //! State variables
   using ParticleBase<Tdim>::state_variables_;
   //! Volumetric mass density (mass / volume)
-  Eigen::Matrix<double, 1, Tnphases> mass_density_;
+  double mass_density_;
   //! Mass
-  Eigen::Matrix<double, 1, Tnphases> mass_;
+  double mass_;
   //! Volume
-  Eigen::Matrix<double, 1, Tnphases> volume_;
+  double volume_;
+  //! Phase
+  unsigned phase_{0};
```

Comment:
I am not sure whether this variable is needed... Could you elaborate on what we can use this for? 
If this is needed to initiate the size of certain variables in nodes or to iterate over the phase inside the particle, should it be better to make it 1 by default for single-phase?
If this is used to call a certain variable, why we need to save it?

Reply:
This is used to set the variable index for mass/momentum of a particular phase in the nodes. So we use a phase index of 0 for single phase. It doesn't have to be defined as a member variable, we can define it inside individual functions.

Reply:
If you have two phase or three, are you going to safe n number of unsigned such as solid_phase, water_phase, or air_phase? I rather define it in each function to call.

Reply:
I have a new definition of phases using an enum, so the `phase_` is removed.

Reply:
I like this one!

Reply:
Let's keep it this way for now.